src/obstacle_infos_publisher.cpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <Eigen/Dense>
+#include <boost/shared_ptr.hpp>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+#include <yaml-cpp/yaml.h>
+#include <ros/ros.h>
+
+namespace panda_torque_mpc {
+
+class ObstacleInfosPublisher {
+  ObstacleInfosPublisher(
+      const std::shared_ptr<ros::NodeHandle> &ph,
+      const boost::shared_ptr<pinocchio::Model> &pin_model_,
+      const boost::shared_ptr<pinocchio::GeometryModel> &collision_model_);
```

Comment:
```suggestion
  ObstacleInfosPublisher(
      const std::shared_ptr<ros::NodeHandle>& nh,
      const boost::shared_ptr<pinocchio::Model>& pin_model,
      const boost::shared_ptr<pinocchio::GeometryModel>& collision_model);
```
Formatting plus in this case it is better to use `nh` instead `ph`, as it is not a private node handle in this case